tests/other/test_smtlibv2.py

```diff
@@ -986,6 +1026,33 @@ def test_check_solver_undefined(self):
         )
         self.assertTrue(self.solver._solver_version() > Version(major=4, minor=4, patch=1))
 
+    def testRelated(self):
+        cs = ConstraintSet()
+        aa1 = cs.new_bool(name="AA1")
+        aa2 = cs.new_bool(name="AA2")
+        bb1 = cs.new_bool(name="BB1")
+        bb2 = cs.new_bool(name="BB2")
+        cs.add(Operators.OR(aa1, aa2))
+        cs.add(Operators.OR(bb1, bb2))
+        self.assertTrue(self.solver.check(cs))
+        # No BB variables related to AA
+        self.assertNotIn("BB", cs.related_to(aa1).to_string())
+        self.assertNotIn("BB", cs.related_to(aa2).to_string())
+        self.assertNotIn("BB", cs.related_to(aa1 == aa2).to_string())
+        self.assertNotIn("BB", cs.related_to(aa1 == False).to_string())
+        # No AA variables related to BB
+        self.assertNotIn("AA", cs.related_to(bb1).to_string())
+        self.assertNotIn("AA", cs.related_to(bb2).to_string())
+        self.assertNotIn("AA", cs.related_to(bb1 == bb2).to_string())
+        self.assertNotIn("AA", cs.related_to(bb1 == False).to_string())
+
+        # Nothing is related to tautologies?
+        self.assertEqual("", cs.related_to(simplify(bb1 == bb1)).to_string())
+
+        # But if the tautollogy can not get simplified we have to ask the solver
+        # and send in all the other stuff
+        self.assertNotIn("AA", cs.related_to(bb1 == bb1).to_string())
+
```

Comment:
I think it'd be nice to check that we _do_ find what we expect in our `related_to` call.

One example
```suggestion

        related_bb1 = cs.related_to(bb1).to_string()
        self.assertIn("BB1", related_bb1)
        self.assertIn("BB2", related_bb1)
```
Since
```python
related == '(declare-fun BB2 () Bool)\n(declare-fun BB1 () Bool)\n(assert (or BB1 BB2))\n'
```

Are there any more interesting cases?

Reply:
I did not want to lock in how it is generated and more in the fact that the unrelated vars are sliced out. Idk.. stuff may be optimized? no in this case.
Actually I do not need to make a string out of it to check if a var is declared.... :thinking:

Reply:
Hmmm. That's a fair point. I'm fine with it now then.